source/rust_verify/src/rust_to_vir_base.rs

```diff
@@ -138,10 +138,17 @@ pub(crate) fn hack_get_def_name<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId) -> Strin
     debug_name[last_colon + 1..].to_string()
 }
 
-pub(crate) fn ident_to_var<'tcx>(ident: &Ident) -> String {
+pub(crate) fn foreign_param_to_var<'tcx>(ident: &Ident) -> String {
     ident.to_string()
 }
 
+pub(crate) fn local_to_var<'tcx>(
+    ident: &Ident,
+    local_id: rustc_hir::hir_id::ItemLocalId,
+) -> String {
+    ident.to_string() + "$$" + &local_id.index().to_string()
```

Comment:
Can we do this operation (string + "$$" + string) in `def.rs` so that we keep track of the "$$" with the other prefixes, suffixes, and separators?  Also, "$$" is used for SUBST_RENAME_SEPARATOR.  Maybe local_to_var could use "~" instead?